### PR TITLE
Update Docker Compose for 4.3-beta

### PR DESCRIPTION
**Pull request by vmstan**
Alternative to #28148 with minimal changes to prepare for 4.3-beta release.

- Changes streaming to new image.
- Updates tags to 4.3
- Converts wget health check commands to curl

### CODE REVIEW

**Review by ClearlyClaire — APPROVED** [review]

**Review by renchap — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -79,16 +79,16 @@ services:
...
     env_file: .env.production
-    command: node ./streaming
+    command: node ./streaming/index.js
     networks:
       - external_network
       - internal_network
     healthcheck:
       # prettier-ignore
-      test: ['CMD-SHELL', 'wget -q --spider --proxy=off localhost:4000/api/v1/streaming/health || exit 1']
+      test: ['CMD-SHELL', "curl -s --noproxy localhost localhost:4000/api/v1/streaming/health | grep -q 'OK' || exit 1'"]
```
Comment (thomas-pike) [inline docker-compose.yml:91]: There seems to be an erroneous `'` in this healthcheck after the `exit 1`

Comment (vmstan) [reply docker-compose.yml:91]: Yes it has been fixed by #31612 for the next beta.

**Comment by melroy89** [comment]
I still get:

```sh
[FATAL tini (7)] exec node failed: No such file or directory
```

With the Mastodon streaming container in Docker using `ghcr.io/mastodon/mastodon:v4.3.0`...

**Comment by melroy89** [comment]
Uh.. Node is not installed in your prebuild images!?? 

image

**Comment by vmstan** [comment]
Please refer to the 4.3 release notes. The streaming container is a separate image.

**Pull request merged**